front: fix spacetimechart not updating

SharglutDev opened the pull request:
See commits description

close #8107

SharglutDev committed:
front: fix spacetimechart not updating

Add a new state fetchSpaceTimeData to control when we want to update the spaceTimeData.
In getSpaceTimeChartData function, as we use .initiate to handle the trainScheduleProjectPath data, we need to unsubscribe to it.

flomonster reviewed (changes requested):
This doesn't fix the bug

SharglutDev committed:
front: fix and clean timestops margins when updating a train

As editoast doesn't accept 'none' as a margin anymore, we can now always display the first margin value for the origin.
As editoast doesn't accept a margin for destination, ensure that its margin is always undefined

flomonster reviewed (changes requested):
It fixes the bug but I found a regression:

1. Create a timetable with multiple trains (having different path)
2. Try to change the projection path (it doesn't refresh the space time chart)

clarani commented:
The issue https://github.com/OpenRailAssociation/osrd/issues/8107 was solved by #8271
